builder.ts

```diff
@@ -289,24 +306,23 @@ export class PackDefinitionBuilder implements BasicPackDefinition {
    * ```
    */
   setSystemAuthentication(systemAuthentication: SystemAuthenticationDef): this {
-    const {
-      getConnectionName: getConnectionNameDef,
-      getConnectionUserId: getConnectionUserIdDef,
-      postSetup: postSetupDef,
-      ...rest
-    } = systemAuthentication;
-    const getConnectionName = wrapMetadataFunction(getConnectionNameDef);
-    const getConnectionUserId = wrapMetadataFunction(getConnectionUserIdDef);
-    const postSetup = postSetupDef?.map(step => {
-      return {...step, getOptions: wrapMetadataFunction(setEndpointDefHelper(step).getOptions)};
-    });
-    this.systemConnectionAuthentication = {
-      ...rest,
-      getConnectionName,
-      getConnectionUserId,
-      postSetup,
-    } as SystemAuthentication;
+    this.systemConnectionAuthentication = this._wrapAuthenticationFunctions(
+      systemAuthentication,
+    ) as SystemAuthentication;
+    return this;
+  }
 
+  /**
+   * @hidden
+   */
+  addUserAuthentication(auxAuth: AuxiliaryAuthentication): this {
```

Comment:
I think it's confusing to have addUserAuthentication and setUserAuthentication. Would lean into auxiliary more and do addAuxiliaryAuthentication.

Reply:
What about overloading `setUserAuthentication` to take in an array of authentication types? Similar to how we extended `BaseAuthentication.networkDomain` to take an array of domains.

Reply:
I think I prefer a separate function because these authentications aren't meant for just _any_ user.